segmentstore/server/host/src/main/java/io/pravega/segmentstore/server/host/stat/AutoScaleMonitor.java

```diff
@@ -34,14 +34,14 @@
     @VisibleForTesting
     public AutoScaleMonitor(@NonNull StreamSegmentStore store, @NonNull EventStreamClientFactory clientFactory,
                             @NonNull AutoScalerConfig configuration) {
-        this.executor = ExecutorServiceHelpers.newScheduledThreadPool(configuration.getThreadPoolSize(), "auto-scaler");
+        this.executor = ExecutorServiceHelpers.newScheduledThreadPool(configuration.getThreadPoolSize(), "auto-scaler", Thread.MIN_PRIORITY);
```

Comment:
Please be aware a min priority thread can be starved.

Reply:
Fully aware of that. But calculating and generating autoscale metrics is of less importance than ingesting data itself. If we reach a situation where our CPUs are fully loaded, then we need to prioritize ingestion over this.

This CPU is mostly used for AutoScaleMonitor/Processor and SegmentStatsImpl cache cleanup and writing events to the Controller's auto scale stream.

In my local tests I have found this thread pool to take up considerable CPU. That is likely because, previously, the code in `SegmentStatsRecorderImpl` was executing one callback per Store append. Now that I've batched them together this should be less of a concern.